community/pike/APKBUILD

```diff
@@ -0,0 +1,235 @@
+	local dn=$(dirname "$1")
+	local dst="$2/$dn"
+	mkdir -p "$dst"
+	mv "$1" "$dst/$bn"
+}
+
+_install() {
+	local bn=$(basename "$1")
+	local dn=$(dirname "$1")
+	for item in `find $dn -maxdepth 1 -name "$bn"`; do
+		_move_item "$item" "$2"
+	done
+}
+
+bzip2() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname bzip2 support"
+	depends="pike-core=$pkgver-r$pkgrel bzip2"
+	_install "usr/lib/pike/modules/*Bz2.*" "$subpkgdir"
+}
+
+dnssd() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname DNS SD support"
+	depends="pike-core=$pkgver-r$pkgrel"
+	_install "usr/lib/pike/modules/*DNS_SD.*" "$subpkgdir"
+}
+
+fuse() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname fUSE support"
+	depends="pike-core=$pkgver-r$pkgrel"
+	_install "usr/lib/pike/modules/*Fuse.so" "$subpkgdir"
+	_install "usr/lib/pike/modules/Fuse.pmod" "$subpkgdir"
+}
+
+gdbm() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname GDBM support"
+	depends="pike-core=$pkgver-r$pkgrel gdbm"
+	_install "usr/lib/pike/modules/*Gdbm.*" "$subpkgdir"
+}
+
+gl() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname OpenGL/GLUT support"
+	depends="pike-core=$pkgver-r$pkgrel freeglut mesa-gl libsm"
+	_install "usr/lib/pike/modules/*GL*" "$subpkgdir"
+}
+
+gtk() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname GTK2 support"
+	depends="pike-image=$pkgver-r$pkgrel libglade gtk+2.0"
+	_install "usr/lib/pike/modules/*GTK2*" "$subpkgdir"
+	_install "usr/lib/pike/modules/Tools.pmod/PV.pike" "$subpkgdir"
+}
+
+image() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname image handling modules"
+	depends="pike-core=$pkgver-r$pkgrel giflib-utils libjpeg tiff libwebp"
+	_install "usr/lib/pike/?.?/modules/Image.pmod" "$subpkgdir"
+	_install "usr/lib/pike/modules/Image.so" "$subpkgdir"
+	_install "usr/lib/pike/modules/*Image_[!S]*.so" "$subpkgdir"
+	_install "usr/lib/pike/modules/*Image*.pmod" "$subpkgdir"
+	_install "usr/lib/pike/modules/Graphics.pmod" "$subpkgdir"
+	_install "usr/lib/pike/modules/Colors.pmod" "$subpkgdir"
+	_install "usr/lib/pike/modules/Protocols.pmod/X.pmod/XImage.pmod" "$subpkgdir"
+}
+
+pcre() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname PCRE support"
+	depends="pike-core=$pkgver-r$pkgrel pcre"
+	_install "usr/lib/pike/modules/*Regexp_PCRE.*" "$subpkgdir"
+}
+
+sdl() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname SDL support"
+	depends="pike-image=$pkgver-r$pkgrel sdl sdl_mixer"
+	_install "usr/lib/pike/modules/*SDL.*" "$subpkgdir"
+}
+
+svg() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname SVG support"
+	depends="pike-core=$pkgver-r$pkgrel librsvg"
+	_install "usr/lib/pike/modules/_Image_SVG.so" "$subpkgdir"
+}
+
+sqlite() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname SQLite support"
+	depends="pike-core=$pkgver-r$pkgrel sqlite-libs"
+	_install "usr/lib/pike/modules/*SQLite.*" "$subpkgdir"
+	_install "usr/lib/pike/modules/Sql.pmod/sqlite*" "$subpkgdir"
+}
+
+doc() {
+	cd "$pkgdir"
+	pkgdesc="$pkgname documentation"
+	depends="pike-core=$pkgver-r$pkgrel"
+	mkdir -p "$subpkgdir/usr/share/"
+	gzip usr/share/man/man1/pike.1
+	mv usr/doc/ "$subpkgdir/usr/share/"
+	mv usr/share/man/ "$subpkgdir/usr/share/"
+}
```

Comment:
Letting the `default_doc` function handle this would be better.

Reply:
The only thing I see that might not be handled by `default_doc` is the `usr/doc` directory. You could change the location of `usr/doc` in the `package` function to `usr/share/doc` so that it is split without any need for your own `doc` implementation.